solution/ui/regulations/eregs-vite/src/components/policy-repository/SubjectSelector.vue

```diff
@@ -15,42 +20,140 @@ const props = defineProps({
 const $router = useRouter();
 const $route = useRoute();
 
+const state = reactive({
+    filter: "",
+    subjects: [],
+});
+
+watch(
+    () => props.policyDocSubjects.loading,
+    (loading) => {
+        if (loading) {
+            state.subjects = [];
+            return;
+        }
+
+        state.subjects = props.policyDocSubjects.results;
+    }
+);
+
+const getFilteredSubjects = (filter) => {
+    if (!filter || filter.length < 1) {
+        state.subjects = props.policyDocSubjects.results;
+        return;
```

Comment:
we use the passed in list of subjects from props if `filter` reactive object entry is falsy